docs/pages/application-access/guides/vnet.mdx

```diff
@@ -0,0 +1,197 @@
+of arbitrary domains for unsuspecting users.
+
+Wait a couple of minutes and verify that the DNS record changes have been propagated.
+
+```code
+$ host -t TXT <Var name="suffix" />
+company.test descriptive text "teleport-cluster=teleport.example.com"
+```
+
+<Admonition type="note" title="DNS TXT and CNAME records">
+If the domain you chose for the suffix has the CNAME record set, you won't be able to add a TXT
+record. A CNAME record must be the only record for the given name. In that scenario, you can set the
+suffix to a domain one level above if possible (e.g., `company.test` instead of
+`internal.company.test`) or switch from CNAME to A records.
+</Admonition>
+
+## Step 3/4. Set `public_addr` of the app
+
+Set `public_addr` field of the application in the Application Service configuration file
+`/etc/teleport.yaml` and restart the `teleport` daemon.
+
+```yaml
+version: v3
+# …
+app_service:
+  # …
+  apps:
+  - name: "tcp-app"
+    uri: tcp://localhost:5432
+    public_addr: "<Var name="public_addr" />"
+```
+
+## Step 4/4. Connect
+
+Once you [start VNet](../../connect-your-client/vnet.mdx), you should be able to connect to the
+application over the custom `public_addr` using the application client you would normally use to
+connect to it. You might need to restart VNet if it was already running while you were making
+changes to the cluster.
+
+```code
+$ psql postgres://postgres@tcp-app.company.test/postgres
+```
+
+## Next steps
+
+### Configuring IPv4 CIDR range
+
+VNet uses `100.64.0.0/10` as the IPv4 CIDR range by default. Each resource is assigned an IP address
+from that range just before the first connection is established. If this range conflicts with other
+software deployed on devices of your users, it can be changed by setting the `ipv4_cidr_range` field
+of the VNet config.
+
+When starting, VNet fetches configs from all available clusters and arbitrarily picks a range
+provided by one of the clusters. This means that if you have a root cluster with multiple leaf
+clusters, you should update the range on each cluster.
+
+Create a file called `vnet-config.yaml`:
+
+```yaml
+kind: vnet_config
+version: v1
+metadata:
+  name: vnet-config
+spec:
+  ipv4_cidr_range: "100.64.0.0/10"
+```
+
+Create the VNet config:
+
+```code
+$ tctl create vnet-config.yaml
+vnet_config has been created
+```
+
+If the config already exists, you can use `tctl edit` instead:
+
+```
+$ tctl edit vnet_config
+```
+
+### Configuring leaf cluster apps
+
+To make a [leaf cluster](../../management/admin/trustedclusters.mdx) app accessible over a custom
+`public_addr`, you need to follow the same steps while being logged in directly to the leaf cluster.
+
+```code
+$ tsh login --proxy=<Var name="leaf.example.com" /> --user=<Var name="email@example.com" />
+```
+
+`public_addr` needs to be set in the Application Service configuration file belonging to the leaf
+cluster and the DNS TXT record needs to contain the name of the leaf cluster.
```

Comment:
might want to mention that you can have multiple TXT records on the same domain if you want to use it in multiple clusters